### DISCUSSION TITLE
Can these tracing gaps be explained?

### DISCUSSION THREAD

**Discussion by wakamoleguy** [prisma/prisma#23301] [Q&A]
### Question

Hello, team! We’re debugging some performance issues in our API code, and I’m hoping somebody can help me explain a gap in the traces from Prisma.

This is a batched operation of about 1,000 findUnique operations. I’ve trimmed all but the last operation from the screenshot below. There are 3 nested fields being requested, hence the 3 prisma:engine:db_query spans.

<img width="1394" alt="Screenshot 2024-02-27 at 18 08 31" src="https://github.com/prisma/prisma/assets/3210372/989efb36-b9fc-41d3-bd06-cf2240485ae6">

What might be happening between the prisma:engine:connection and prisma:engine:db_query spans? I’m not so familiar with the engine code, so any ideas on where to look would be helpful. Anybody run into similar issues?

Also, it looks like there is a similar gap in the handoff between the prisma:client and prisma:engine spans. If there is a slowdown caused by the number of operations being batched, I wonder if it is affecting both gaps in the trace? Not sure if that would make sense.

### Information about Prisma Schema, Client Queries and Environment (optional)

Our schema is quite large, but here are the models we are querying:

```prisma
generator client {
  provider        = "prisma-client-js"
  output          = "../../../node_modules/.prisma/client"
  previewFeatures = ["metrics", "tracing"]
}

generator server_types {
  provider = "prisma-client-js"
  output   = "../node_modules/.prisma/client"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}


model Position {
  id                   Int                       @id @default(autoincrement())
  name                 String
  description          String?
  descriptionUrl       String?
  level                Int
  createdAt            DateTime                  @default(now())
  updatedAt            DateTime                  @updatedAt
  ladderId             Int
  organizationId       Int
  type                 PositionType              @default(ANNUAL)  
  authorId             Int?
  jobCodes             String[]
  author               User?                     @relation(fields: [authorId], references: [id])
  ladder               Ladder                    @relation(fields: [ladderId], references: [id], onDelete: Cascade)
  organization         Organization              @relation(fields: [organizationId], references: [id], onDelete: Cascade)
  cashBands            CashBand[]
  employments          Employment[]
  equityBands          EquityBand[]
  offerContents        OfferContent[]
  recItems             RecItem[]
  levelingCodeMappings PositionLevelingMapping[]
  employeeLevelings    EmployeeLeveling[]

  @@index([ladderId])
  @@index([organizationId])
}


model Ladder {
  id             Int          @id @default(autoincrement())
  name           String
  description    String?
  createdAt      DateTime     @default(now())
  updatedAt      DateTime     @updatedAt
  departmentId   Int
  organizationId Int
  authorId       Int?
  author         User?        @relation(fields: [authorId], references: [id])
  department     Department   @relation(fields: [departmentId], references: [id], onDelete: Cascade)
  organization   Organization @relation(fields: [organizationId], references: [id], onDelete: Cascade)
  positions      Position[]

  @@index([departmentId])
  @@index([organizationId])
}


model Department {
  id             Int          @id @default(autoincrement())
  name           String
  description    String?
  createdAt      DateTime     @default(now())
  organizationId Int
  updatedAt      DateTime     @updatedAt
  organization   Organization @relation(fields: [organizationId], references: [id], onDelete: Cascade)
  ladders        Ladder[]

  @@index([organizationId])
}
```

```ts
async function getPosition(
  ctx: Context,
  positionId: number
) {
  const organizationId = currentOrgId(ctx);

  const position = await ctx.prisma.position.findUnique({
    where: { id: positionId, organizationId },
    include: { ladder: { include: { department: true } } },
  });

  if (!position) return null;

  return position;
}
```

- OS: Docker node:18-buster-slim
- Database: PostgreSQL
- Node.js version: 18.7.0

- Run `prisma -v` to see your Prisma version and paste it
(This is from my local environment; the trace was not taken from Mac OS)
```
prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : darwin
Operating System        : darwin
Architecture            : x64
Node.js                 : v18.7.0
Query Engine (Node-API) : libquery-engine 5a9203d0590c951969e85a7d07215503f4672eb9 (at ../../node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Schema Engine           : schema-engine-cli 5a9203d0590c951969e85a7d07215503f4672eb9 (at ../../node_modules/@prisma/engines/schema-engine-darwin)
Schema Wasm             : @prisma/prisma-schema-wasm 5.10.0-34.5a9203d0590c951969e85a7d07215503f4672eb9
Default Engines Hash    : 5a9203d0590c951969e85a7d07215503f4672eb9
Studio                  : 0.499.0
Preview Features        : metrics, tracing
```

**Comment by janpio** [comment]
`tracing` being a preview feature, we might just be missing some data. Is this reproducible in an independent project that you could share with us, so we can reproduce? Then you can just create a bug issue about this so we can look into it, please.

**Reply by wakamoleguy** [reply]
I'll see if I can create a reproduction in a separate repo, thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
